event_routing_backends/processors/xapi/transformer.py

```diff
@@ -0,0 +1,92 @@
+"""
+xAPI Transformer Class
+"""
+from tincan import Agent, LanguageMap, Statement, Verb
+
+from event_routing_backends.helpers import get_anonymous_user_id_by_username
+from event_routing_backends.processors.mixins.base_transformer import BaseTransformerMixin
+from event_routing_backends.processors.xapi import constants
+
+
+class XApiTransformer(BaseTransformerMixin):
+    """
+    xAPI Transformer Class
+    """
+    required_fields = (
+        'object',
+        'verb',
+    )
+
+    def transform(self):
+        """
+        Return transformed `Statement` object.
+
+        `BaseTransformer`'s `transform` method will return dict containing
+        xAPI objects in transformed fields. Here we return a `Statement` object
+        constructed using those fields.
+
+        Returns:
+            `Statement`
+        """
+        transformed_props = super(XApiTransformer, self).transform()
+        return Statement(**transformed_props)
+
+    def base_transform(self):
+        """
+        Transform the fields that are common for all events.
+        """
+        # TODO: Move context registration in base transform
+        self.transformed_event = {
+            'actor': self.get_actor(),
+            'timestamp': self.get_timestamp()
+        }
+
+    def get_actor(self):
+        """
+        Return `Agent` object for the event.
+
+        Returns:
+            `Agent`
+        """
+        return Agent(
+            openid=get_anonymous_user_id_by_username(self.event['context'].get('username')),
```

Comment:
@feanil I was validating generated statements and found that openid is a URI that uniquely identifies the Agent(user) and here we are using UUID. Is there a way to generate a unique URI for user identification?

Reply:
Let's use `https://openedx.org/users/user-v1:<uuid>` instead.  I believe the existing xAPI integration in enterprise is doing something like this already.

Reply:
done